includes/polling/functions.inc.php

```diff
@@ -626,17 +630,26 @@ function json_app_get($device, $extend, $min_version = 1)
 {
     $output = snmp_get($device, 'nsExtendOutputFull.' . string_to_oid($extend), '-Oqv', 'NET-SNMP-EXTEND-MIB');
 
+    // save for returning if not JSON
+    $orig_output = $output;
+
     // make sure we actually get something back
     if (empty($output)) {
         throw new JsonAppPollingFailedException('Empty return from snmp_get.', -2);
     }
 
+    // checks for base64 decoding and converts it to non-base64 so it can gunzip
+    if (preg_match('/^[A-Za-z0-9\/\+\n]+\=*\n*$/', $output)) {
+        $output = base64_decode($output);
+        $output = gzdecode($output);
+    }
+
     //  turn the JSON into a array
     $parsed_json = json_decode(stripslashes($output), true);
 
     // improper JSON or something else was returned. Populate the variable with an error.
     if (json_last_error() !== JSON_ERROR_NONE) {
-        throw new JsonAppParsingFailedException('Invalid JSON', $output, -3);
+        throw new JsonAppParsingFailedException('Invalid JSON', $orig_output, -3);
```

Comment:
Why orig_output?  base64 encoded string won't be very helpful.

Reply:
```
    // improper JSON or something else was returned. Populate the variable with an error.
    if (json_last_error() !== JSON_ERROR_NONE) {
        throw new JsonAppParsingFailedException('Invalid JSON', $orig_output, -3);
    }
```

Because decoding it may be a mistake given the expect operation of this function supporting returning the raw return upon a JSON decoding failure for fall back non-JSON processing via the poller.

That said not aware of thing currently that could vaguely be mistaken as looking like BASE64 for when it comes to any of the pollers that have older possible non-JSON data handling. This is more for being future friendly.

Also lets say there is a extend that gets created that produces some really simple new line seperated data, this could easily be mistaken for BASE64. Then some time down the line it gets converted to use JSON and then this function, on the off chance it properly decades the base64 and then amazingly ungzips it, it would then fail at this point and return corrupted data.

That said will want to do return that as well when it comes to implementing the checks that @Jellyfrog brought up. Will need to add in a two new exceptions as well for that. :)